### PR TITLE
Update to rust 1.14 nightly and update the example dependencies

### PR DESCRIPTION
**Pull request by est31**
- Add a gitignore to make working with the repo easier
- Update the example to use the latest sdl. Also update its Cargo.toml to reflect the change.
- Remove use of the deprecated clock_ticks crate.

I have tested the example, and playback of webm is working well.

### CODE REVIEW

**Comment by est31** [comment]
Also I wonder whether the nightly branch should maybe be retired? It seems to have lots of duplicate work, and as servo updates very often it might not be needed anymore. Maybe we could even try to work towards targetting stable, then its guaranteed that servo compiles.

**Comment by est31** [comment]
Note that I haven't checked the mac OS build, as I don't have a mac.

File: codecs/libavcodec.rs [hunk 1]
```diff
@@ -335,13 +335,14 @@ impl AvFrame {
...
         unsafe {
-            slice::from_raw_mut_buf(&(*self.frame).data[channel], len as usize)
+            slice::from_raw_parts(*&(*self.frame).data[channel], len as usize)
         }
     }
 }
 
 pub struct AvPacket<'a> {
     packet: ffi::EitherAVPacket,
+    phantom: ::std::marker::PhantomData<&'a[u8]>,
```
Comment (pcwalton) [inline codecs/libavcodec.rs]: uber-nit: put a space before `[u8]`
Also I think I'd prefer just importing `PhantomData` at the top of the file.

File: codecs/libavcodec.rs [hunk 2]
```diff
@@ -472,13 +474,13 @@ impl videodecoder::VideoDecoder for VideoDecoderImpl {
...
         let mut data: Vec<_> = data.iter().map(|x| *x).collect();
-        for _ in range(0, FF_INPUT_BUFFER_PADDING_SIZE) {
+        for _ in 0..FF_INPUT_BUFFER_PADDING_SIZE {
             data.push(0);
         }
 
         let mut packet = AvPacket::new(data.as_mut_slice());
-        let presentation_time = *presentation_time;
-        self.context.borrow_mut().set_get_buffer_callback(Box::new(|frame: &AvFrame| {
+        let presentation_time = presentation_time.clone();
```
Comment (pcwalton) [inline codecs/libavcodec.rs:483]: Why do you need to explicitly clone this? Isn't it `Copy`?

Comment (est31) [reply codecs/libavcodec.rs:483]: I'm getting a lifetime error when I remove that line. I also get it when I write `let presentation_time = presentation_time`. clone works.

File: codecs/libavcodec.rs (hunk 1 above)
Comment (est31) [reply codecs/libavcodec.rs]: okay, changed both

**Pull request merged**